app/Http/Controllers/AdminController.php

```diff
@@ -376,57 +376,6 @@ public function upgrade()
-            self::delete_unused_rows('dailyupdate', 'projectid', 'project');
-            self::delete_unused_rows('subproject', 'projectid', 'project');
-            self::delete_unused_rows('coveragefilepriority', 'projectid', 'project');
-            self::delete_unused_rows('user2project', 'projectid', 'project');
-            self::delete_unused_rows('userstatistics', 'projectid', 'project');
-
-            self::delete_unused_rows('build2configure', 'buildid', 'build');
-            self::delete_unused_rows('build2note', 'buildid', 'build');
-            self::delete_unused_rows('build2test', 'buildid', 'build');
-            self::delete_unused_rows('buildemail', 'buildid', 'build');
-            self::delete_unused_rows('builderror', 'buildid', 'build');
-            self::delete_unused_rows('builderrordiff', 'buildid', 'build');
-            self::delete_unused_rows('buildfailure', 'buildid', 'build');
-            self::delete_unused_rows('buildfailuredetails', 'id', 'buildfailure', 'detailsid');
-            self::delete_unused_rows('buildinformation', 'buildid', 'build');
-            self::delete_unused_rows('buildtesttime', 'buildid', 'build');
-            self::delete_unused_rows('configure', 'id', 'build2configure', 'configureid');
-            self::delete_unused_rows('configureerror', 'configureid', 'configure');
-            self::delete_unused_rows('configureerrordiff', 'buildid', 'build');
-            self::delete_unused_rows('coverage', 'buildid', 'build');
-            self::delete_unused_rows('coveragefilelog', 'buildid', 'build');
-            self::delete_unused_rows('coveragesummary', 'buildid', 'build');
-            self::delete_unused_rows('coveragesummarydiff', 'buildid', 'build');
-            self::delete_unused_rows('dynamicanalysis', 'buildid', 'build');
-            self::delete_unused_rows('label2build', 'buildid', 'build');
-            self::delete_unused_rows('subproject2build', 'buildid', 'build');
-            self::delete_unused_rows('summaryemail', 'buildid', 'build');
-            self::delete_unused_rows('testdiff', 'buildid', 'build');
-
-            self::delete_unused_rows('dynamicanalysisdefect', 'dynamicanalysisid', 'dynamicanalysis');
-            self::delete_unused_rows('subproject2subproject', 'subprojectid', 'subproject');
-
-            self::delete_unused_rows('dailyupdatefile', 'dailyupdateid', 'dailyupdate');
-            self::delete_unused_rows('coveragefile', 'id', 'coverage', 'fileid');
-            self::delete_unused_rows('coveragefile2user', 'fileid', 'coveragefile');
-
-            self::delete_unused_rows('dailyupdatefile', 'dailyupdateid', 'dailyupdate');
-            self::delete_unused_rows('test2image', 'outputid', 'testoutput');
-
-            $xml .= add_XML_value('alert', 'Database cleanup complete.');
-        }
-
         /* Check the builds with wrong date */
         if ($CheckBuildsWrongDate) {
```

Comment:
Are these two if-blocks still useful at this point?  This feels like something we should do automatically, or at least do via the CLI.

Reply:
I doubt anyone is using those buttons. I would be okay with removing them or converting them to Artisan commands if they're still needed.